src/main/java/uk/gov/ch/developer/docs/controller/developer/SignInController.java

```diff
@@ -7,26 +7,21 @@
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
-import uk.gov.ch.oauth.IOauth;
-import uk.gov.ch.oauth.identity.IIdentityProvider;
+import uk.gov.ch.oauth.IOAuthCoordinator;
 
 @Controller
 @RequestMapping("${signin.url}")
 public class SignInController {
 
 
     @Autowired
-    private IOauth oauth;
-
-    @Autowired
-    IIdentityProvider identityProvider;
+    private IOAuthCoordinator coordinator;
 
 
     @GetMapping
     public void doSignIn(final HttpServletRequest httpServletRequest,
                          final HttpServletResponse httpServletResponse) throws IOException {
-        final String state = oauth.prepareState(httpServletRequest);
-        final String authoriseUri = identityProvider.getAuthorisationUrl(state);
+        final String authoriseUri = coordinator.getAuthoriseUriFromRequest(httpServletRequest);
         httpServletResponse.sendRedirect(authoriseUri);
```

Comment:
I would prefer that we use "redirect:"+authoriseUri;
so that we don't need to worry about the IOException and we can leave spring to handle it.